.github/workflows/test.yaml

```diff
@@ -117,16 +118,14 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: build
-        shell: bash
-        run: |
-          PACKAGE_NAME_PREFIX="${{ github.workflow }}"
-          if [ "${{ github.event_name }}" = "pull_request" ]; then
-            PACKAGE_NAME_PREFIX="$PACKAGE_NAME_PREFIX-${{ github.event.number }}"
-          fi
-          PACKAGE_NAME_PREFIX="$PACKAGE_NAME_PREFIX-${{ github.sha }}"
-          export PACKAGE_NAME_PREFIX
```

Comment:
It's unfortunate to lose this. In my version of the release system, I had this in mind when I set it up so a custom version could be specified via an environment variable:
- https://github.com/arduino/arduino-lint/blob/7af3d443215c8b3e08614aadf60df666f6a1f31b/DistTasks.yml#L197
- https://github.com/arduino/arduino-lint/blob/7af3d443215c8b3e08614aadf60df666f6a1f31b/.github/workflows/nightly.yml#L27

Any chance we could do something like that?

Reply:
What do you think of something like this?

```diff
diff --git a/Taskfile.yml b/Taskfile.yml
index a2c4f605..8f4d5faa 100755
--- a/Taskfile.yml
+++ b/Taskfile.yml
@@ -213,7 +213,7 @@ vars:
     sh: echo "{{now | date "20060102"}}"
   TAG:
     sh: echo "`git tag --points-at=HEAD 2> /dev/null`"
-  VERSION: "{{ if .NIGHTLY }}nightly-{{ .TIMESTAMP_SHORT }}{{ else if .TAG }}{{ .TAG }}{{ else }}snapshot{{ end }}"
+  VERSION: "{{ if .NIGHTLY }}nightly-{{ .TIMESTAMP_SHORT }}{{ else if .TAG }}{{ .TAG }}{{ else }}{{ .PACKAGE_NAME_PREFIX }}snapshot{{ end }}"
   LDFLAGS: >
     -ldflags
     '
````

```diff
diff --git a/.github/workflows/test.yaml b/.github/workflows/test.yaml
index 3c81dcf2..e1e1aef5 100644
--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -125,7 +119,14 @@ jobs:
           version: 3.x
 
       - name: Build
-        run: task dist:all
+        run: |
+          PACKAGE_NAME_PREFIX="${{ github.workflow }}-"
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            PACKAGE_NAME_PREFIX="$PACKAGE_NAME_PREFIX-${{ github.event.number }}-"
+          fi
+          PACKAGE_NAME_PREFIX="$PACKAGE_NAME_PREFIX-${{ github.sha }}-"
+          export PACKAGE_NAME_PREFIX
+          task dist:all
 
       # Uploads all architectures as separate artifacts
       - name: Upload Linux 32 bit artifact
```

I remember removing this because it was kind of a pain to test but thinking about it I just have to change the test like this:

```diff
diff --git a/test/test_main.py b/test/test_main.py
index e9c1a092..6f171fa5 100644
--- a/test/test_main.py
+++ b/test/test_main.py
@@ -38,7 +38,7 @@ def test_version(run_command):
     parsed_out = json.loads(result.stdout)
     assert parsed_out.get("Application", False) == "arduino-cli"
     version = parsed_out.get("VersionString", False)
-    assert semver.VersionInfo.isvalid(version=version) or version == "snapshot" or "nightly" in version
+    assert semver.VersionInfo.isvalid(version=version) or "snapshot" in version or "nightly" in version
     assert isinstance(parsed_out.get("Commit", False), str)
```

Reply:
Looks good!